hls4ml/model/optimizer/passes/conv_to_convxd.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+
+from hls4ml.model.layers import Constant, Conv, Conv1D, Conv2D
+from hls4ml.model.optimizer import OptimizerPass
+
+# these are attributes to copy
+_base_attributes = (
+    'in_width',
+    'out_width',
+    'n_chan',
+    'n_filt',
+    'pad_left',
+    'pad_right',
+    'filt_width',
+    'stride_width',
+    'dilation_width',
+    'in_height',
+    'out_height',
+    'pad_top',
+    'pad_bottom',
+    'filt_height',
+    'stride_height',
+    'dilation_height',
+    'data_format',
+)
+
+
+class ConvToConvXD(OptimizerPass):
+    """Convert Conv with constant to a Conv1D or Conv2D layer"""
+
+    def match(self, node):
+        is_match = isinstance(node, Conv) and (
+            (len(node.inputs) == 2 and isinstance(node.get_input_node(node.inputs[1]), Constant))
+            or (
+                len(node.inputs) == 3
+                and isinstance(node.get_input_node(node.inputs[1]), Constant)
+                and isinstance(node.get_input_node(node.inputs[2]), Constant)
+            )
+        )
+
+        return is_match
+
+    def transform(self, model, node):
+        """Convert Conv with constant to a Conv1D or Conv2D layer"""
+
+        weight_node = node.get_input_node(node.inputs[1])
+        weight_data = weight_node.attributes['value']
+        bias_node = None
+        if len(node.inputs) == 3:
+            bias_node = node.get_input_node(node.inputs[2])
+
+        # creating the attributes
+        attributes = {k: node.attributes.get(k, None) for k in _base_attributes}
+
+        # The ConvxD nodes expect the weight data to be in a different format, not (M, k1.., C)
+        if node.attributes['n_dim'] == 1:
+            newtype = Conv1D
+            attributes['weight_data'] = np.transpose(weight_data, (1, 2, 0))
```

Comment:
Out of curiosity, why do we need to transpose the weights here when the model is supposed to have been processed by `qonnx-to-channels-last` already?

Reply:
I think the weight data is still not in the same order in ONNX as expected by hls4ml, which follows the Keras order. It's not the channels index that is moved but the feature map index. Let me know if it's not correct, because it was a bit found by experimentation.

Reply:
I am not very familiar with how this is handled in QONNX, but at first glance I don't see the weights being transposed in https://github.com/fastmachinelearning/qonnx/blob/main/src/qonnx/transformation/channels_last.py, so you are probably right. I just don't understand the logic in QONNX then :-D